core/lib/zksync_core/src/reorg_detector/mod.rs

```diff
@@ -21,8 +21,8 @@ use crate::{
 mod tests;
 
 #[derive(Debug, thiserror::Error)]
-enum HashMatchError {
-    #[error("RPC error calling main node")]
+pub enum Error {
+    #[error("RPC error calling main node: {0}")]
```

Comment:
Nit: AFAIU, the wrapped error info is displayed automatically for `{:?}` and `{:#}` presentations; that's why a similar example in [`thiserror` crate docs](https://docs.rs/thiserror/latest/thiserror/#example) doesn't include `: {0}` for the `Disconnect` variant.

Reply:
done